packages/backend/src/managers/inference/inferenceManager.ts

```diff
@@ -383,11 +392,12 @@ export class InferenceManager extends Publisher<InferenceServer[]> implements Di
               port: !!containerInfo.Ports && containerInfo.Ports.length > 0 ? containerInfo.Ports[0].PublicPort : -1,
             },
             status: containerInfo.Status === 'running' ? 'running' : 'stopped',
-            models: modelsId.map(id => this.modelsManager.getModelInfo(id)),
+            models: models,
```

Comment:
I think we can avoid having all this try catch with the `getModelInfo` potentially throwing error by filtering the models Id using the `isModelOnDisk` from the `ModelsManager`

```suggestion
            models: modelsId.filter(id => this.modelsManager.isModelOnDisk(id)).map(id => this.modelsManager.getModelInfo(id)),,
```

Reply:
Your suggestion does a different thing but i don't know which should be the correct behavior tbh.

In the current code i skip the containers that are associated to a model which is not on the catalog/disk anymore.
In your case you're still detecting the inference server but its models list can be empty.

So not sure how it should work. Should we not show the inference servers that have been created with a no-more-existing model or we should show them and associate an empty models list? Not sure if the latter implies some misbehavior though.

Reply:
Agree, opened https://github.com/containers/podman-desktop-extension-ai-lab/issues/791

Reply:
SO for now I would rather have the servers showed even tho we don't have the model, so the user can do cleanup from ai studio

Reply:
Done